pytesmo/validation_framework/adapters.py

```diff
@@ -39,40 +39,93 @@
 class MaskingAdapter(object):
     """
     Transform the given class to return a boolean dataset given the operator
-    and threshold. This class calls the callse the read_ts and read methods
+    and threshold. This class calls the read_ts and read methods
     of the given instance and applies boolean masking to the returned data
     using the given operator and threshold.
 
     Parameters
     ----------
     cls: object
-        has to have read_ts method
+        has to have a read_ts or read method
     operator: string
-        one of '<', '<=', '==', '>=', '>'
-    threshold: float
+        one of '<', '<=', '==', '>=', '>', '!='
+    threshold:
         value to use as the threshold combined with the operator
-
+    column_name: string, optional
+        name of the column to cut the read masking dataset to
     """
 
-    def __init__(self, cls, op, threshold):
+    def __init__(self, cls, op, threshold, column_name = None):
         self.cls = cls
 
         self.op_lookup = {'<': operator.lt,
                           '<=': operator.le,
                           '==': operator.eq,
+                          '!=': operator.ne,
                           '>=': operator.ge,
                           '>': operator.gt}
         self.operator = self.op_lookup[op]
         self.threshold = threshold
 
+        self.column_name = column_name
+
+    def __mask(self, data):
+        if self.column_name is not None:
+            data = data.loc[:, [self.column_name]]
+        return self.operator(data, self.threshold)
+
     def read_ts(self, *args, **kwargs):
         data = self.cls.read_ts(*args, **kwargs)
-        return self.operator(data, self.threshold)
+        return self.__mask(data)
 
     def read(self, *args, **kwargs):
         data = self.cls.read(*args, **kwargs)
-        return self.operator(data, self.threshold)
+        return self.__mask(data)
+
+class SelfMaskingAdapter(object):
+    """
+    Transform the given (reader) class to return a dataset that is masked based 
+    on the given column, operator, and threshold. This class calls the read_ts 
+    or read method of the given reader instance, applies the operator/threshold
+    to the specified column, and masks the whole dataframe with the result.
+
+    Parameters
+    ----------
+    cls: object
+        has to have a read_ts or read method
+    operator: string
+        one of '<', '<=', '==', '>=', '>', '!='
+    threshold:
+        value to use as the threshold combined with the operator
+    column_name: string
+        name of the column to cut the read masking dataset to
```

Comment:
I guess this should be something like `name of the column to apply the threshold to`?

Reply:
Yes, that sounds better (although the given column will be the only one remaining in the DataFrame after masking).

Reply:
Are we talking about the changes in MaskingAdapter or about the SelfMaskingAdapter?

Reply:
SelfMaskingAdapter. Line 101 of the diff.

Reply:
Oh, sorry, my bad. I've changed it.